### PR TITLE
pixelfed: 0.11.5 -> 0.11.6

### PR DESCRIPTION
**Pull request by RaitoBezarius**
###### Description of changes

This bring many fixes for PostgreSQL and mobile APIs.

Testing on my own infra.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/servers/web-apps/pixelfed/composer-env.nix [hunk 1]
```diff
@@ -4,16 +4,11 @@
...
 let
   inherit (phpPackages) composer;
-
-  filterSrc = src:
-    builtins.filterSource (path: type: type != "directory" || (baseNameOf path != ".git" && baseNameOf path != ".git" && baseNameOf path != ".svn")) src;
-
   buildZipPackage = { name, src }:
     stdenv.mkDerivation {
       inherit name src;
-      nativeBuildInputs = [ unzip ];
```
[the remark this answers is not in the document]
Comment (RaitoBezarius) [reply pkgs/servers/web-apps/pixelfed/composer-env.nix:14]: I think I used a master with some fixes but that's not impossible

**Comment by wlcx** [comment]
Worth updating this PR to 0.11.8? Would resolve https://github.com/NixOS/nixpkgs/issues/235147
Happy to lend a hand!

**Comment by RaitoBezarius** [comment]
> Worth updating this PR to 0.11.8? Would resolve #235147 Happy to lend a hand!

For sure, apologies, I was quite busy. If I go AWOL, don't hesitate to steal this PR, apply changes, request my review and Sandro's and we will get this merged.

**Comment by RaitoBezarius** [comment]
What is really annoying is that composer2nix is out of tree and really in bad shape :<.

**Comment by imsofi** [comment]
Do you know if there is any good reason why composer2nix could not be pulled into nixpkgs? It seems like a strange omission for such a large language as PHP.

I see that https://github.com/NixOS/nixpkgs/pull/198032 is one attempt.

**Comment by imsofi** [comment]
From some deeper digging, [a proper PHP builder for nixpkgs](https://github.com/NixOS/nixpkgs/pull/225401) is currently being developed, and likely the future instead of pulling in `composer2nix` to nixpkgs like i commented above.

**Comment by RaitoBezarius** [comment]
> Do you know if there is any good reason why composer2nix could not be pulled into nixpkgs? It seems like a strange omission for such a large language as PHP.
> 
> I see that #198032 is one attempt.

I think it's the high level of un-maintenance unfortunately…

**Comment by drupol** [comment]
I just succeeded to build successfully pixelfed with the new PHP builder, here's what the derivation look like:

```nix
{ lib
, fetchFromGitHub
, php
, nixosTests
, dataDir ? "/var/lib/pixelfed"
, runtimeDir ? "/run/pixelfed"
}:

php.buildComposerProject (finalAttrs: {
  pname = "pixelfed";
  version = "0.11.5";

  # GitHub distribution does not include vendored files
  src = fetchFromGitHub {
    owner = "pixelfed";
    repo = "pixelfed";
    rev = "v${finalAttrs.version}";
    hash = "sha256-ZrvYKMSx5WymWR46/UKr5jCsclXXzBeY21ju22zeqN0=";
  };

  vendorHash = "sha256-Naav15Sr7JkhVAyOzJKQgcULyFMMX2R1SmuaZ/ymty4=";

  postInstall = ''
    rm -R $out/share/php/pixelfed/bootstrap/cache
    # Move static contents for the NixOS module to pick it up, if needed.
    mv $out/share/php/pixelfed/bootstrap $out/share/php/pixelfed/bootstrap-static
    mv $out/share/php/pixelfed/storage $out/share/php/pixelfed/storage-static
    ln -s ${dataDir}/.env $out/share/php/pixelfed/.env
    ln -s ${dataDir}/storage $out/share/php/pixelfed/
    ln -s ${dataDir}/storage/app/public $out/share/php/pixelfed/public/storage
    ln -s ${runtimeDir} $out/share/php/pixelfed/bootstrap
    chmod +x $out/share/php/pixelfed/artisan
  '';

  passthru.tests = { inherit (nixosTests) pixelfed; };

  meta = with lib; {
    description = "A federated image sharing platform";
    license = licenses.agpl3Only;
    homepage = "https://pixelfed.org/";
    maintainers = with maintainers; [ raitobezarius ];
    platforms = php.meta.platforms;
  };
})
```

The PHP Builder PR is looking for some fresh eyes and people to review it. It's been more than 2 months that I'm working on it and a some new eyes on it would be very welcome.

**Pull request closed without merging**